Homework 12. JS DOM

DmitriyF89 opened the pull request:
Task-01.
Function addClassVisible(current, list) accepts a list of elements and a position of the element to be displayed.
For the "current" element it add class "visible", for all other elements class "visible" is deleted.
Silder has eventListener which works when a button is clicked. Callback function reevaluates the "current" element and invokes addClassVisible function with new current item.

Task-02
Click on the accountButton toggles display property of the userAcc to none/block values, and also toggles class "rotate180" to arrow image.

DmitriyF89 committed:
feat: upload completed tasks of Homework js-dom

DmitriyF89 committed:
BREAKING CHANGE(task-02): change the way to integrate user-acc menu

fix(task-01): add class names to elements to exclude tag selectors
fix(task-02): change class names, css structure, and some minor fixes in the markup

aneelia commented on src/ex12_js_dom/src/task-01/img/witch.svg:
```diff
@@ -0,0 +1,69 @@
+    </rdf:description>
+  </rdf:rdf>
+</metadata>
+<g transform="translate(0.000000,300.000000) scale(0.100000,-0.100000)" fill="#060606" stroke="none">
```
может попробовать через онлайн-нормалайзер какой нить пропустить свг?

DmitriyF89 replied:
Прогнал через https://www.svgminify.com/

DmitriyF89 committed:
chore: optimise SVG file

aneelia commented on src/ex12_js_dom/task-02.html:
```diff
@@ -0,0 +1,70 @@
+
+  <script src="./task-02.js"></script>
+</body>
+
```
пустые строки убрать надо бы

DmitriyF89 replied:
Удалил пустую строку перед script.
Между body и html строки сами добавляются (плагин автоматически их добавляет при сохранении =)). Так можно оставить или тоже нужно удалять?

aneelia commented on src/ex12_js_dom/task-02.js:
```diff
@@ -0,0 +1,31 @@
+const
+  accountButton = document.querySelector('.login'),
+  loginIconArrow = document.querySelector('.login__arrow-down'),
```
arrow я думаю, тк она имеет два состояния

DmitriyF89 replied:
Оставил .login__arrow

aneelia commented on src/ex12_js_dom/task-02.js:
```diff
@@ -0,0 +1,31 @@
+  </div>
+  `;
+
+let isUserAccActive = false;
```
логчнее isMenuOpen

DmitriyF89 replied:
Переименовал в isMenuOpen

DmitriyF89 committed:
style: naming fixes in task-02.html and task-02.js

DmitriyF89 committed:
fix: change class name of arrow icon in task-02.html

DmitriyF89 committed:
BREAKING CHANGE: Rework Kanban view to fullscreen version.

style: small rework in css
rework function addClassVisible rework with using forEach loop (task-01)
add removing eventListeners on window.unload event.

DmitriyF89 committed:
style: minor fixes
